backend/app/libs/tasks/redis.py

```diff
@@ -1,10 +1,32 @@
+import json
 from typing import Any, Union
 
 import redis
 
 BALANCES_KEY_TEMPLATE = "{address}_{symbol}"
 
 
+def store_troublesome_treasuries(
+    troublesome_treasuries: list[tuple[str, int]],
+    provider: Union[redis.Redis, redis.client.Pipeline],
+):
+    troublesome_treasury_payload = [
+        json.dumps({"address": address, "chain_id": chain_id})
```

Comment:
This line builds an object with shape `{address, chain_id}`, which is stored in Redis. Then on line 25, you are loading this object and deconstructing it.  
This means the object is never used by code outside of this module. Thus, this is a purely internal abstraction.  
You could reduce this module's complexity by just storing the json dump of the `(address, chain_id)` tuples as passed in argument.

For simplicity's sake, I would even recommend letting go of the `chain_id` value. Your functions will be reduced to one-liners.